proto/src/common/node.rs

```diff
@@ -22,7 +22,8 @@ pub enum NodeKind {
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Metrics {
-    pub cpu_usage: f64,
+    /// Collection of CPU usages.
+    pub cpu_usage: Vec<f64>,
```

Comment:
I'd also prefer a f64 as a starting point... Using (and sending) a vec seems really overkill for what we're proposing, even though I can see the motivations behind it.

Reply:
I think we should stick with the `Vec`, representing the CPUs usages as a single f64 is *very* inaccurate and could lead to performance issues such as ignoring `workers` that are capable of running a task just because one of it's CPUs is being highly used. WDYT?

Reply:
Tbh, for the sake of closing this PR so we can move on to other components, let's keep it simple by now and just use a `f64`, let time tell if that `Vec` will be a real need.